### ISSUE TITLE
Validator PubKey Validation Inconsistent With Other Clients

### ISSUE THREAD

**Issue by kanewallmann** [Consensys-Incorporated/teku#4787]
Teku's validation of validator pub keys in `/eth/v1/beacon/states/{state_id}/validators` is inconsistent with other clients.

If you make a call to the above endpoint with one or more invalid pub keys in the `id` parameter such as `0x000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001`, Teku responds with the following error:

```
{"status":400,"message":"Invalid public key: 0x000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001"}
```

Other major clients ignore pub keys that don't match an existing validator regardless of whether the pub key is valid or not. The [Eth Beacon Node API](https://ethereum.github.io/beacon-APIs/#/Beacon/getStateValidators) specifies `If an index or public key does not match any known validator, no information will be returned but this will not cause an error.`. While it doesn't explicitly state that invalid pub keys should be ignored it can be interpreted that way and indeed other client implementation (perhaps unintentionally) took that interpretation.

Other clients tested: Nimbus, Prysm, Lighthouse.

**Comment by Neurone** [comment]
The Eth Beacon Node API technically allows to return `400` for `Invalid state or validator ID, or status`, so I think the other clients must comply to this behavior.

A fix by Teku side is however about the content of the response: `"status":400` should be changed in `"code":400`.

So, currently no client is compliant with the specs :smile:

**Comment by rolfyone** [comment]
Thanks for mentioning that departure from standard @Neurone, I've addressed that issue, but it should be noted if you're looking at 'status' currently, that'll change next release.

**Comment by rolfyone** [comment]
Did a little testing.
validator id of `-1` throws a 400, much the same as the invalid pubkey above, where as a pubkey thats valid and not found will just not be added to the list, so i think the behaviour is consistent with that.
I've raised a spec issue on the api and will see if there's a lot of discussion there about it. I think that now that we've fixed the response it's probably ok as it stands, but it's just a matter of if there's traction on clarifying the standard api to describe 'invalid id'
